plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/MavenProxyFacet.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-2015 Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.maven.internal;
+
+import java.io.IOException;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.sonatype.nexus.common.time.Clock;
+import org.sonatype.nexus.repository.content.InvalidContentException;
+import org.sonatype.nexus.repository.proxy.ProxyFacetSupport;
+import org.sonatype.nexus.repository.view.Context;
+import org.sonatype.nexus.repository.view.Payload;
+
+import org.joda.time.DateTime;
+
+import static com.google.common.base.Preconditions.checkState;
+
+/**
+ * Maven proxy facet that talks to {@link MavenFacet} and {@link MavenTransportFacet}.
+ *
+ * @since 3.0
+ */
+@Named
+public class MavenProxyFacet
+    extends ProxyFacetSupport<MavenPayload>
+{
+  private static final String NOW = MavenProxyFacet.class.getName() + ".now";
+
+  private final Clock clock;
+
+  private MavenFacet mavenFacet;
+
+  private MavenTransportFacet mavenTransportFacet;
+
+  @Inject
+  public MavenProxyFacet() {
+    this.clock = new Clock();
```

Comment:
And more complex to add a ctor here to otherwise set a private final to a new instance